### DISCUSSION TITLE
Return 3010 ERROR_SUCCESS_REBOOT_REQUIRED

### DISCUSSION THREAD

**Discussion by thxkiwi** [wixtoolset/issues#7435] [Questions]
I'm wanting to put together a WiX installer that installs INFs. The installer could be invoked from the command line or programmatically by another UI. Assuming use of pnputil within the installer to perform the actual INF installation, what is the recommended method for the installer itself to return 3010 (ERROR_SUCCESS_REBOOT_REQUIRED) without actually rebooting, if and only if, pnputil returns 3010?

**Comment by robmen** [comment]
Not thinking too deeply about it, you'd probably need to wrap wrap `pnputil` with a DLL custom action since non-zero exit codes are considered failures for EXE custom actions. Maybe this is something QuietExec could be enhanced to do? I don't know, I haven't investigated deeplly.

**Reply by thxkiwi** [reply]
@robmen thanks for the reply and suggestions.

We could certainly use a custom action and then communicate the pnputil return value back through a property via the Session. What I'm not clear on is given return value is available after the custom action, how can we make the WiX generated installer actually exit with that value?
